### DISCUSSION TITLE
Advice on changing database location

### DISCUSSION THREAD

**Discussion by Flier2632** [immich-app/immich#14963] [Q&A] [answered]
Hello,

I've had immich working successfully for about 2 weeks (docker compose).

In my **.env** file, **DB_DATA_LOCATION** is set to **./postgres** as standard.

> PUID=1000
> PGID=1000
> TZ="Europe/London"
> USERDIR="/home/user"
> DOCKERDIR="/home/user/docker"
> DATADIR="/media/downloads"
> MYSQL_ROOT_PASSWORD=**********
> GUACAMOLE_ROOT_PASSWORD=************
> PIHOLE_WEB_PASSWORD=****************
> 
> ################## Immich Entries
> UPLOAD_LOCATION=/media/immichdata
> DB_DATA_LOCATION=./postgres
> #/ The Immich version to use. You can pin this to a specific version like "v1.71.0"
> IMMICH_VERSION=release
> DB_PASSWORD=*************
> #/ The values below this line do not need to be changed
> DB_USERNAME=postgres
> DB_DATABASE_NAME=immich

This means that there's a postgres folder created in the folder where my various containers live.

![image](https://github.com/user-attachments/assets/a29e81fb-72ed-4ab0-a12a-71d7afe80089)

I'd like to edit this, so that the location is more like **/home/user/docker/immich_postgres**. This is the usual convention for my existing containers.

![image](https://github.com/user-attachments/assets/e766502d-e203-484d-b41b-a262fe8ba65a)

I believe I can create the folders by editing **DB_DATA_LOCATION=$DOCKERDIR/appdata/immich_postgres/**.
However, I also believe that while that would create the folder, it wouldn't migrate the data from the existing location and get it all working cleanly.

Can you please advise as to how I'd go about this with minimal service interruption?

**Comment by bo0tzz** [comment, marked as answer]
You need to copy the folder yourself. Since this is on the same system just `cp` should be fine. Note that if you're migrating between different systems it's best to do a full backup/restore.

**Reply by Flier2632** [reply]
So, are you saying:

1. Create the new folder,
2. Copy files from old to new folder,
3. Restart containers.

And all should be well?

**Reply by bo0tzz** [reply]
Start by stopping the containers, and you'll want to change the path in .env of course. Also (and as always) make sure you have good backups.

**Reply by Flier2632** [reply]
Many thanks @bo0tzz.

I did as you said and all is well.
